### PR TITLE
Ustream live wip

### PR DESCRIPTION
**Pull request by jasongornall**

### CODE REVIEW

File: src/Player.js [hunk 1]
```diff
@@ -75,6 +75,7 @@ export default class Player extends Component {
       const playedSeconds = this.getCurrentTime() || 0
       const loadedSeconds = this.getSecondsLoaded()
       const duration = this.getDuration()
+
```
Comment (andybarron) [inline src/Player.js]: probs should get rid of this so our diffs aren't muddied

File: src/Player.js [hunk 2]
```diff
@@ -84,10 +85,14 @@ export default class Player extends Component {
           progress.loadedSeconds = loadedSeconds
           progress.loaded = loadedSeconds / duration
         }
-        // Only call onProgress if values have changed
-        if (progress.played !== this.prevPlayed || progress.loaded !== this.prevLoaded) {
+
+        // Special Case for Live Typed so they still OnProgress
```
Comment (andybarron) [inline src/Player.js]: What Even Is proper Capitalization ;)

Comment (andybarron) [reply src/Player.js]: also `typed` -> `types`?

File: test/specs/ReactPlayer.js [hunk 3]
```diff
@@ -93,6 +93,16 @@ const TEST_URLS = [
     url: 'https://www.mixcloud.com/mixcloud/meet-the-curators/',
     switchTo: 'https://www.mixcloud.com/mixcloud/mixcloud-curates-4-mary-anne-hobbs-in-conversation-with-dan-deacon/',
     skip: true
+  },
+  {
```
Comment (jasongornall) [inline test/specs/ReactPlayer.js:97]: can probably remove this

File: test/specs/ReactPlayer.js [hunk 4]
```diff
@@ -249,23 +260,23 @@ describe('ReactPlayer', () => {
...
-          url: test.url,
-          onDuration: d => { duration = d },
-          onProgress: p => {
-            if (!seeked && duration && p.playedSeconds > 1) {
-              player.seekTo(duration - 1)
-              seeked = true
-            }
-          },
-          onEnded: () => done()
+      if (test.onSeek) {
```
Comment (jasongornall) [inline test/specs/ReactPlayer.js:257]: I guss onSeek isn't seekTo I think it's fine tho

File: test/specs/ReactPlayer.js [hunk 5]
```diff
@@ -154,6 +164,7 @@ describe('ReactPlayer', () => {
         // Use a count object to ensure everything is called at least once
         let count = {}
         const bump = key => {
+          console.log(key)
```
Comment (jasongornall) [inline test/specs/ReactPlayer.js]: remove

Comment (andybarron) [inline test/specs/ReactPlayer.js]: 😿

File: src/players/index.js [hunk 6]
```diff
@@ -7,8 +7,9 @@ import { Wistia } from './Wistia'
 import { Twitch } from './Twitch'
 import { DailyMotion } from './DailyMotion'
 import { Mixcloud } from './Mixcloud'
-import { FilePlayer } from './FilePlayer'
+import { UstreamLive } from './UstreamLive'
```
Comment (andybarron) [inline src/players/index.js]: remove empty line

File: src/players/index.js [hunk 7]
```diff
@@ -7,8 +7,9 @@ import { Wistia } from './Wistia'
...
 
+import { FilePlayer } from './FilePlayer'
 export default [
```
Comment (andybarron) [inline src/players/index.js]: add empty line above

**Review by andybarron — CHANGES_REQUESTED** [review]

**Pull request merged**